open_rarity/resolver/opensea_api_helpers.py

```diff
@@ -257,87 +350,136 @@ def get_collection_with_metadata_from_opensea(
-    and stores it in the Collection object
+    and stores it in the Collection object. If local cache file is used and
+    contains all collection token data, that data will be used instead.
 
     Parameters
     ----------
-    opensea_collection_slug : str
+    slug : str
         collection slug on opensea's system
 
     batch_size: int
         batch size for the opensea API requests
         maximum value is 30
 
+    use_cache: bool
+        set to True to look for a local cached version of the collection and token
+        metadata fetched from opensea to prevent re-fetching.
+
     Returns
     -------
     Collection
         collection abstraction
 
     """
     # Fetch collection metadata
-    collection_obj = fetch_opensea_collection_data(
-        slug=opensea_collection_slug
-    )
+    collection_obj = fetch_opensea_collection_data(slug=slug)
     contracts = collection_obj["primary_asset_contracts"]
     interfaces = set([contract["schema_name"] for contract in contracts])
     stats = collection_obj["stats"]
-    if not interfaces.issubset(set(["ERC721", "ERC1155"])):
+    if not interfaces.issubset(set(["ERC721"])):
         raise ERCStandardError(
-            "We currently do not support non EVM standards at the moment"
+            "We currently do not support non ERC721 standards at the moment"
         )
 
     total_supply = int(stats["total_supply"])
+    tokens = get_all_collection_tokens(
+        slug=slug,
+        total_supply=total_supply,
+        batch_size=batch_size,
+        use_cache=use_cache,
+        cache_file_prefix=cache_file_prefix,
+    )
 
-    # Fetch token metadata
-    tokens: list[Token] = []
-    num_batches = math.ceil(total_supply / batch_size)
-    initial_token_id = 0
-
-    # Returns a list of `batch_size` token IDs, such that no token ID
-    # can exceed `max_token_id` (in which case len(return_value) < `batch_size`)
-    def get_token_ids(
-        batch_id: int, max_token_id: int = total_supply - 1
-    ) -> list[int]:
-        token_id_start = initial_token_id + (batch_id * batch_size)
-        token_id_end = int(min(token_id_start + batch_size - 1, max_token_id))
-        return list(range(token_id_start, token_id_end + 1))
-
-    for batch_id in range(num_batches):
-        token_ids = get_token_ids(batch_id)
-        tokens_batch = get_tokens_from_opensea(
-            opensea_slug=opensea_collection_slug, token_ids=token_ids
+    return Collection(name=collection_obj["name"], tokens=tokens)
+
+
+def write_collection_data_to_file(
+    filename: str, slug: str, tokens: list[Token]
+):
+    json_output = []
+    for token in tokens:
+        # Note: We assume EVM token here
+        json_output.append(
+            {
+                "contract_address": token.token_identifier.contract_address,
+                "token_id": token.token_identifier.token_id,
+                "metadata_dict": token.metadata.to_attributes(),
+            }
```

Comment:
nit: you could also implement a `to_dict` and `from_dict` function for `Token` and `TokenIdentifier` so this code can be chain/token standard agnostic.

Reply:
Originally didn't want to add since felt like it wasn't needed but changed my mind - added.